reconciler/incremental.go

```diff
@@ -242,12 +232,24 @@ func (round *incrementalRound[Obj]) commitStatus() {
 		// Update the object if it is unchanged. It may happen that the object has
 		// been updated in the meanwhile, in which case we ignore the status as the
 		// update will be picked up by next reconciliation round.
-		round.table.CompareAndSwap(wtxn, result.rev, round.config.SetObjectStatus(obj, result.status))
-
-		if result.status.Kind == StatusKindError {
-			// Reconciling the object failed, so add it to be retried now that its
-			// status is updated.
-			round.retries.Add(obj, result.rev, false)
+		var status Status
+		if result.err == nil {
+			status = StatusDone()
+		} else {
+			status = StatusError(result.err)
+			numErrors++
+		}
+		_, _, err := round.table.CompareAndSwap(wtxn, result.rev,
+			round.config.SetObjectStatus(obj, status))
+
+		if result.err != nil && err == nil {
+			// Reconciliation of the object had failed and the status was updated
+			// successfully (object had not changed). Queue the retry for the object.
+			newRevision := round.table.Revision(wtxn)
+			round.retries.Add(result.original.(Obj), newRevision, false, result.err)
+		} else if result.err != nil && err != nil {
+			fmt.Printf("FAIL queued %v for retry due to %s ! %s\n", result.original, result.err, err)
```

Comment:
Should this be a log? or a panic?

Reply:
Neither. It was debug logging that I forgot there. Removed!